sdk/mobile-center-distribute/src/test/java/com/microsoft/azure/mobile/distribute/DistributeBeforeApiSuccessTest.java

```diff
@@ -251,6 +320,108 @@ public void accept(Void aVoid) {
+        clickListener.getValue().onClick(mock(Dialog.class), DialogInterface.BUTTON_POSITIVE);
+
+        /* Since we were disabled, no action but toast to explain what happened. */
+        verify(mToast).show();
+    }
+
+    @Test
+    public void disableBeforeHandleUpdateSetupFailureDialogReinstallAction() throws Exception {
+        when(PreferencesStorage.getString(PREFERENCE_KEY_UPDATE_SETUP_FAILED_MESSAGE_KEY)).thenReturn("failed_message");
+
+        /* Trigger call. */
+        start();
+        Distribute.getInstance().onActivityResumed(mock(Activity.class));
+
+        /* Verify dialog. */
+        ArgumentCaptor<DialogInterface.OnClickListener> clickListener = ArgumentCaptor.forClass(DialogInterface.OnClickListener.class);
+        verify(mDialogBuilder).setCancelable(false);
+        verify(mDialogBuilder).setTitle(R.string.mobile_center_distribute_update_failed_dialog_title);
+        verify(mDialogBuilder).setMessage("failed_message");
+        verify(mDialogBuilder).setNegativeButton(eq(R.string.mobile_center_distribute_update_failed_dialog_reinstall), clickListener.capture());
+        verify(mDialog).show();
+
+        /* Disable. */
+        Distribute.setEnabled(false);
+        clickListener.getValue().onClick(mock(Dialog.class), DialogInterface.BUTTON_NEGATIVE);
+
+        /* Since we were disabled, no action but toast to explain what happened. */
+        verify(mToast).show();
+    }
+
+    @Test
+    public void handleFailedUpdateSetupDialogReinstallOption() throws URISyntaxException {
+        when(PreferencesStorage.getString(PREFERENCE_KEY_UPDATE_SETUP_FAILED_MESSAGE_KEY)).thenReturn("failed_message");
+        when(mDialog.isShowing()).thenReturn(false);
+        when(mDialogBuilder.create()).thenReturn(mDialog);
+
+        /* Trigger call. */
+        start();
+        Distribute.getInstance().onActivityResumed(mock(Activity.class));
+
+        /* Verify dialog. */
+        ArgumentCaptor<DialogInterface.OnClickListener> clickListener = ArgumentCaptor.forClass(DialogInterface.OnClickListener.class);
+        verify(mDialogBuilder).setCancelable(false);
+        verify(mDialogBuilder).setTitle(R.string.mobile_center_distribute_update_failed_dialog_title);
+        verify(mDialogBuilder).setMessage("failed_message");
+        verify(mDialogBuilder).setNegativeButton(eq(R.string.mobile_center_distribute_update_failed_dialog_reinstall), clickListener.capture());
+        verify(mDialog).show();
+        verifyStatic();
+        PreferencesStorage.remove(PREFERENCE_KEY_UPDATE_SETUP_FAILED_MESSAGE_KEY);
+        when(BrowserUtils.appendUri(anyString(), anyString())).thenThrow(new URISyntaxException("ex", "message"));
+
+        /* Click. */
+        clickListener.getValue().onClick(mDialog, DialogInterface.BUTTON_NEGATIVE);
+        verifyStatic();
+        PreferencesStorage.remove(PREFERENCE_KEY_UPDATE_SETUP_FAILED_PACKAGE_HASH_KEY);
+    }
+
+    @Test
+    @PrepareForTest(DistributeUtils.class)
```

Comment:
Are we using the real implementation of DistributeUtils in other methods of this file? If no we could move it the PrepareForTest to class level.

Reply:
I like this idea, thanks for pointing!